### PR TITLE
platform-checks: Add a Backup+Restore scenario

### PR DESCRIPTION
**Pull request by philip-stoev**
### Motivation

Backup+Restore was only merged with a minimal test. It needs to be made part of as many test frameworks as possible.

### CODE REVIEW

File: ci/nightly/pipeline.template.yml [hunk 1]
```diff
@@ -337,6 +337,28 @@ steps:
               composition: platform-checks
               args: [--scenario=RestartEntireMz, "--seed=$BUILDKITE_JOB_ID"]
 
+      - id: checks-backup-restore-before-manipulate
+        label: "Backup + restore before manipulate(#2)"
```
Comment (nrainer-materialize) [inline ci/nightly/pipeline.template.yml:341]: Optional: Can we replace `(#2)` with something more expressive?

File: misc/python/materialize/checks/backup_actions.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+from typing import TYPE_CHECKING
+
+from materialize.checks.actions import Action
+from materialize.checks.executors import Executor
+from materialize.mzcompose.services.minio import MINIO_BLOB_URI
+
+if TYPE_CHECKING:
```
Comment (nrainer-materialize) [inline misc/python/materialize/checks/backup_actions.py:16]: Why do we need this?

File: misc/python/materialize/checks/backup_actions.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    pass
+
+
+class Backup(Action):
+    def execute(self, e: Executor) -> None:
+        c = e.mzcompose_composition()
+        c.exec("mc", "mc", "mb", "persist/crdb-backup")
+        c.exec(
+            "cockroach",
+            "cockroach",
```
Comment (nrainer-materialize) [inline misc/python/materialize/checks/backup_actions.py:26]: (I find this a bit special but ok 😉. It was also not changed within this PR.)

**Review by nrainer-materialize — APPROVED** [review]

File: misc/python/materialize/checks/backup_actions.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+from materialize.mzcompose.services.minio import MINIO_BLOB_URI
+
+if TYPE_CHECKING:
+    pass
+
+
+class Backup(Action):
+    def execute(self, e: Executor) -> None:
+        c = e.mzcompose_composition()
+        c.exec("mc", "mc", "mb", "persist/crdb-backup")
```
Comment (def-) [inline misc/python/materialize/checks/backup_actions.py:23]: mc mc mb :D

**Review by def- — APPROVED** [review]
> Backup+Restore was only merged with a minimal test. It needs to be made part of as many test frameworks as possible.

Good callout, adding to my list for parallel-workload and data-ingest where it makes sense as a new scenario too.

File: misc/python/materialize/checks/backup_actions.py (hunk 2 above)
Comment (philip-stoev) [reply misc/python/materialize/checks/backup_actions.py:16]: copy and paste from another file that needed this to satisfy the mypy type checker -- without it the type checker could not type check and python complained about a circular dependency.

File: misc/python/materialize/checks/backup_actions.py (hunk 4 above)
Comment (philip-stoev) [reply misc/python/materialize/checks/backup_actions.py:23]: mc = container name
mc = binary name
mb = "make bucket" command line argument :-)

File: misc/python/materialize/checks/backup_actions.py (hunk 3 above)
Comment (philip-stoev) [reply misc/python/materialize/checks/backup_actions.py:26]: @nrainer-materialize by "special" do you mean the way SQL is being sent to cockroach?

File: ci/nightly/pipeline.template.yml (hunk 1 above)
Comment (philip-stoev) [reply ci/nightly/pipeline.template.yml:341]: The best I can do is say "between the two manipulate() phases" ?

Comment (nrainer-materialize) [reply ci/nightly/pipeline.template.yml:341]: Ok, I think that gets too verbose

File: misc/python/materialize/checks/backup_actions.py (hunk 3 above)
Comment (nrainer-materialize) [reply misc/python/materialize/checks/backup_actions.py:26]: No, I am not too happy about the signature of the invoked method. It apparently often causes the first two arguments to be the same and looks a bit confusing (also see the comment "mc mc mb :D" by def-). I think comprehensibility might be better if `args` (second and following args) were an array but consider that out of scope for this PR.

Comment (philip-stoev) [reply misc/python/materialize/checks/backup_actions.py:26]: oh ok understood. let me think about this. right now they are args to `docker compose exec` but you want them to be args over `docker compose exec $service_name ...` 

What I wanted to be able to do is `mc_service.create_bucket()` and `cockroach_service.run_sql()` , so I may try to implement something like that in a follow-up refactor.

**Pull request merged**